modules/@apostrophecms/seo-fields-global/index.js

```diff
@@ -42,5 +72,38 @@ module.exports = {
         group
       }
       : null;
+  },
+  apiRoutes(self) {
+    return {
+      get: {
+        '/robots.txt': async (req) => {
+          const criteria = {
+            type: '@apostrophecms/global'
+          };
+          try {
+            const globalDoc = await self.apos.doc.find(req, criteria).toObject();
+            let robotsTxtContent = '';
+            switch (globalDoc.robotsTxtSelection) {
+              case 'allow':
+                robotsTxtContent = 'User-agent: *\nDisallow: ';
+                break;
+              case 'disallow':
+                robotsTxtContent = 'User-agent: *\nDisallow: /';
+                break;
+              case 'custom':
+                robotsTxtContent = globalDoc.robotsCustomText || 'User-agent: *\nDisallow: /';
```

Comment:
This could be surprising. I think you should go with `required: true` on the custom textarea instead. It won't really be "required" unless the if condition is met.

Reply:
Added.